lib/modules/platform/github/index.ts

```diff
@@ -166,14 +170,26 @@ export async function initPlatform({
     }
   }
   logger.debug({ platformConfig, renovateUsername }, 'Platform config');
-  const platformResult: PlatformResult = {
+  const gitHubRules: HostRule[] = [];
+  if (!platformConfig.isGhe) {
+    gitHubRules.push({
+      matchHost: 'ghcr.io',
+      username: 'dummy',
+      password: token,
+    });
+    gitHubRules.push({
+      matchHost: 'pkg.github.com',
```

Comment:
please validate that all those registries support bearer auth. maybe some need to use basic auch with Username and password like ghcr

Reply:
you are right, this needs more tests for each of them, 
look at this: https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-rubygems-registry 
requires user name + bearer token.

ill test them 1 by 1

Reply:
ruby tested:
 "https://rubygems.pkg.github.com/aaronjensen/api/v1/dependencies (GET,200)": 1

Reply:
maven tested:
 "https://maven.pkg.github.com/gorohash/intellij-idea-maven-reproduction/dev/gorohash/intellij-idea-maven-reproduction/maven-metadata.xml (GET,200)": 1,

Reply:
i needed to add registry URLs rule to maven (settings.xml isn't in the repository) so its not 100% automatic 

@viceice i think they all work the same, 
im unfamiliar with nuget at all it will waste alot of time for me, can you test nuget using the package u mentioned https://github.com/visualon/MimeMapping/pkgs/nuget/VisualOn.MimeMapping?

Reply:
i think nuget needs basic auth with username and password too
https://github.com/renovatebot/renovate/blob/645f5d519493d8cd8d59dbe3ed0dc0a2290ea5c9/lib/modules/manager/nuget/artifacts.ts#L52-L57

Reply:
no it doesnt, token was enough see:
```
       "https://nuget.pkg.github.com/visualon/download/visualon.mimemapping/2.1.1/visualon.mimemapping.nuspec (GET,404)": 1,
         "https://nuget.pkg.github.com/visualon/index.json (GET,200)": 1,
         "https://nuget.pkg.github.com/visualon/visualon.mimemapping/index.json (GET,200)": 1

```

here's the PR opened 
https://github.com/StinkyLord/nuget-automatic-renovate-auth/pull/2

Reply:
but lockfile update won't work with token. see linked code above

Reply:
i see, username = 'dummy' will work for lockfile update?
it works for      "https://nuget.pkg.github.com/visualon/visualon.mimemapping/index.json (GET,200)": 1

Reply:
yes, i think so

Reply:
Nuget is the only exception, i've checked npm/maven/ruby, i cant even connect to them using username+password

Reply:
will this work? 
```
 if (!platformConfig.isGhe) {
    gitHubRules.push({
      matchHost: 'ghcr.io',
      username: 'dummy',
      password: token,
    });
    gitHubRules.push({
      matchHost: 'pkg.github.com',
      token,
    });
    gitHubRules.push({
      matchHost: 'nuget.pkg.github.com',
      username: 'dummy',
      password: token,
    });
  }
```
i mean does it keep order?